golang/cosmos/x/vbank/module.go

```diff
@@ -106,55 +107,51 @@ func (am AppModule) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.V
 	events := ctx.EventManager().GetABCIEventHistory()
 	addressToBalance := make(map[string]sdk.Coins, len(events)*2)
 
-	ensureBalanceIsPresent := func(address string) error {
-		if _, ok := addressToBalance[address]; ok {
-			return nil
+	// records that we want to emit an balance update for the address
+	// for the given denoms. We use the Coins only to track the set of
+	// denoms, not for the amounts.
+	ensureBalanceIsPresent := func(address string, denoms sdk.Coins) {
+		if denoms.IsZero() {
+			return
 		}
-		account, err := sdk.AccAddressFromBech32(address)
-		if err != nil {
-			return err
+		currentDenoms := sdk.NewCoins()
+		if coins, ok := addressToBalance[address]; ok {
+			currentDenoms = coins
 		}
-		coins := am.keeper.GetAllBalances(ctx, account)
-		addressToBalance[address] = coins
-		return nil
+		addressToBalance[address] = currentDenoms.Add(denoms...)
```

Comment:
Maybe rename `addressToBalance` to `addressToDenoms`, since you're tracking the denoms and the first transferred amount to fill out the balances later.  Also `ensureBalanceIsPresent` to `ensureDenomIsPresent`.

Reply:
I took a look at the usage in context, and `addressToUpdate` and `ensureAddressUpdate` felt best. Changed.